course_catalog/serializers.py

```diff
@@ -387,6 +386,58 @@ def to_internal_value(self, data):
         return super().to_internal_value(course_fields)
 
 
+class OCWNextSerializer(CourseSerializer):
+    """
+    Serializer for creating Course objects from ocw next data
+    """
+
+    def to_internal_value(self, data):
+        """
+        Custom function to parse data out of the raw ocw json
+        """
+
+        topics = [
+            topic for topic_sublist in data.get("topics", []) for topic in topic_sublist
+        ]
+
+        topics = list(set(topics))
+
+        topics = [{"name": topic_name} for topic_name in topics]
+
+        department_numbers = [data.get("primary_course_number", "").split(".")[0]]
+
+        extra_course_numbers = data.get("extra_course_numbers", None)
+
+        if extra_course_numbers:
+            extra_course_numbers = extra_course_numbers.split(", ")
+            for extra_course_number in extra_course_numbers:
+                department_number = extra_course_number.split(".")[0]
+                if department_number not in department_numbers:
+                    department_numbers.append(department_number)
+        else:
+            extra_course_numbers = []
+
+        course_fields = {
+            "course_id": f"{data.get('uid')}+{data.get('primary_course_number')}",
+            "title": data.get("course_title"),
+            "short_description": data.get("course_description"),
```

Comment:
Minor difference in this field value between old and new import, not sure how important it is.  Plone import contains HTML tags, ocw-next import does not but seems to have some markdown (`<em>` ->` _`) .  Examples below are for "System Dynamics II" (15.872)

On the front end, the markdown is visible (`_15.871_`)

Imported from plone:
```
<p>Continuation of <em>15.871</em>, emphasizing 
```

Imported from ocw-next:
```
Continuation of _15.871_, emphasizing 
```
<img width="439" alt="Screen Shot 2022-01-21 at 9 10 33 AM" src="https://user-images.githubusercontent.com/187676/150541261-eddf8190-ed78-4007-b35e-2641fc1769bc.png">

<img width="439" alt="Screen Shot 2022-01-21 at 9 09 53 AM" src="https://user-images.githubusercontent.com/187676/150541263-f5796635-93d6-4e6c-9f32-33f5a350e71a.png">